lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -364,56 +405,55 @@ def exec_module(self, **kwargs):
             results = nic_to_dict(nic)
             self.log(results, pretty_print=True)
 
+            nsg = None
             if self.state == 'present':
-
+                # check for update
                 update_tags, results['tags'] = self.update_tags(results['tags'])
                 if update_tags:
                     changed = True
 
-                if self.private_ip_address:
-                    if results['ip_configuration']['private_ip_address'] != self.private_ip_address:
-                        self.log("CHANGED: network interface {0} private ip".format(self.name))
-                        changed = True
-                        results['ip_configuration']['private_ip_address'] = self.private_ip_address
-
-                if self.public_ip_address_name:
-                    if results['ip_configuration']['public_ip_address'].get('id') != pip.id:
-                        self.log("CHANGED: network interface {0} public ip".format(self.name))
-                        changed = True
-                        results['ip_configuration']['public_ip_address']['id'] = pip.id
-                        results['ip_configuration']['public_ip_address']['name'] = pip.name
-
                 if self.security_group_name:
-                    if results['network_security_group'].get('id') != nsg.id:
+                    nsg = self.get_security_group(self.security_group_name)
+                    if nsg and results['network_security_group'].get('id') != nsg.id:
                         self.log("CHANGED: network interface {0} network security group".format(self.name))
                         changed = True
-                        results['network_security_group']['id'] = nsg.id
-                        results['network_security_group']['name'] = nsg.name
 
-                if self.private_ip_allocation_method:
-                    if results['ip_configuration']['private_ip_allocation_method'] != self.private_ip_allocation_method:
-                        self.log("CHANGED: network interface {0} private ip allocation".format(self.name))
-                        changed = True
-                        results['ip_configuration']['private_ip_allocation_method'] = self.private_ip_allocation_method
-                        if self.private_ip_allocation_method == 'Dynamic':
-                            results['ip_configuration']['private_ip_address'] = None
+                if results['ip_configurations'][0]['subnet']['virtual_network_name'] != self.virtual_network_name:
```

Comment:
Wouldn't you need to do this check on all the ip_configurations and not just the 1st?